### DISCUSSION TITLE
I don't get all the commits

### DISCUSSION THREAD

**Discussion by nricciardi** [gitpython-developers/GitPython#1589] [Q&A]
I have a test repo where there is master branch. From master I have created branch2 and branch3, then from branch2 I have created branch2-2. Then I have merged branch3 into branch2.

I have created a function to create a recursive dataclasses to represent repository, so then I can display it using GitGraphJS, **BUT** I don't get all the commits when I use `iter_commits` function in a master branch, why?

In addition, I would have the branch name of each commit, but when I use `rev_name` on commit it display stranger name with `~`.

**Comment by Byron** [comment]
I recommend setting up [debug logging](https://gitpython.readthedocs.io/en/stable/tutorial.html#git-command-debugging-and-customization) to see which commands GitPython executes. These can then be executed by hand to see if it truly is GitPython that not listing some commits, or if it is `git` itself producing unexpected output.

**Reply by nricciardi** [reply]
I have fixed it using a for which iterates for each branch and gets commit of the particular branch.

Is there a function to get all commits like a tree instead of list of commits?
As list of commits I must parse they and the process is so slow...

**Reply by Byron** [reply]
No. If performance is an issue, `pygit2` might help, or caching, or using a different programming language entirely.
